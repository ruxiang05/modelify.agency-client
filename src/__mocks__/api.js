const jobMock = {
  status: 'in progress',
  _id: '5c97b9cd0328e6b2aa76a361',
  model: '5c686eb24cb1b33456b0be59',
  agent: '5c686e904cb1b33456b0be58',
  title: 'Test',
  date: '2019-05-26T00:00:00.000Z',
  address: 'address',
  pay: 100,
  description: 'some description',
};

const api = {
  jobs: {
    getJobs: jest.fn(() => Promise.resolve({
      jobs: [jobMock],
    })),
    acceptJob: jest.fn(() => Promise.resolve({
      job: jobMock,
    })),
    completeJob: jest.fn(() => Promise.resolve({
      job: {
        ...jobMock,
        status: 'complete',
      },
    })),
    declineJob: jest.fn(() => Promise.resolve({
      job: {
        ...jobMock,
        status: 'declined',
      },
    })),
    addJob: jest.fn(() => Promise.resolve()),
  },
  googleCalendar: {
    addEvent: jest.fn(() => Promise.resolve()),
    authorize: jest.fn(() => Promise.resolve({ message: 'success', token: 'test' })),
    getAuthURL: jest.fn(() => Promise.resolve({ redirect: '/redirect' })),
  },
  agents: {
    getModels: jest.fn(() => Promise.resolve({
      agent: {
        agentInfo: {
          models: [
            {
              _id: '5ca24d217736960017cae87c',
              agentInfo: { models: [] },
              email: 'jane@doe.com',
              modelInfo: {
                address: '',
                characteristics: {
                  eyes: '',
                  hair: '',
                  skin: '',
                  dateOfBirth: '',
                },
                measurements: {
                  chest: '',
                  height: '',
                  hips: '',
                  waist: '',
                  weight: '',
                },
                name: 'Jane Doe',
                phoneNumber: '07423360809',
                role: 'model',
              },
            },
          ],
        },
      },
    })),
    addModel: jest.fn(() => Promise.resolve()),
  },
  chats: {
    getMessages: jest.fn(() => Promise.resolve({
      messages: [
        {
          message: 'hi',
          user: '1',
        },
        {
          message: 'hello',
          user: '2',
        },
      ],
    })),
    getChats: jest.fn(() => Promise.resolve({
      chats: [
        {
          id: '2t3457345',
          job: {},
          messages: [],
          users: ['1', '2'],
        },
      ],
    })),
  },
  users: {
    signup: jest.fn(() => Promise.resolve({ ok: true })),
    login: jest.fn(() => Promise.resolve({ token: 'abc' })),
    editProfile: jest.fn(() => Promise.resolve({ token: 'abc' })),
  },
};

module.exports = api;
