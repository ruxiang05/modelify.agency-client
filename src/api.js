const url = process.env.REACT_APP_API_URL;

const api = {
  users: {
    signup: data => fetch(`${url}/users/signup`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      }),
    login: data => fetch(`${url}/users/login`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      }),
    editProfile: (token, data) => fetch(`${url}/users/edit`, {
      method: 'PUT',
      body: JSON.stringify(data),
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  agents: {
    getModels: token => fetch(`${url}/agents/`, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    addModel: (token, data) => fetch(`${url}/agents/add-model`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  jobs: {
    addJob: (token, data) => fetch(`${url}/jobs/create`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    getJobs: token => fetch(`${url}/jobs/all`, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    acceptJob: (token, data) => fetch(`${url}/jobs/accept`, {
      method: 'PUT',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    completeJob: (token, data) => fetch(`${url}/jobs/complete`, {
      method: 'PUT',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    declineJob: (token, data) => fetch(`${url}/jobs/decline`, {
      method: 'PUT',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  chats: {
    getChats: token => fetch(`${url}/chats/`, {
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    getMessages: (token, id) => fetch(`${url}/chats/getMessages?id=${id}`, {
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  googleCalendar: {
    authorize: (token, data) => fetch(`${url}/google-calendar/authorize`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    getAuthURL: token => fetch(`${url}/google-calendar/getAuthURL`, {
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    addEvent: (token, data) => fetch(`${url}/google-calendar/addEvent`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
};

export default api;
