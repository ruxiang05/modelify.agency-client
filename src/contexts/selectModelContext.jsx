/* Uses methods and/or components from react */
import React from 'react';

export const SelectModel = React.createContext({
  selectModel: () => {},
});
export const SelectModelConsumer = SelectModel.Consumer;
export const SelectModelProvider = SelectModel.Provider;
