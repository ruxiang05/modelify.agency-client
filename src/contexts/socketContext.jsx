/* Uses methods and/or components from react */
import React from 'react';

export const SocketContext = React.createContext({
  socket: {},
});
export const SocketConsumer = SocketContext.Consumer;
export const SocketProvider = SocketContext.Provider;
