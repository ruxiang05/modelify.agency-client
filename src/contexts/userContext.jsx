/* Uses methods and/or components from react */
import React from 'react';

export const UserContext = React.createContext({
  user: {},
  updateUser: () => {},
});
export const UserConsumer = UserContext.Consumer;
export const UserProvider = UserContext.Provider;
