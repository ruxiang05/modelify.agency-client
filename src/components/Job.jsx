import React from 'react';
import PropTypes from 'prop-types';
import PageHeader from './PageHeader';
import { ReactComponent as DateIcon } from '../assets/icons/calendar-day.svg';
import { ReactComponent as LocationIcon } from '../assets/icons/map-marker-alt.svg';
import { ReactComponent as NoteIcon } from '../assets/icons/sticky-note.svg';
import { ReactComponent as PoundIcon } from '../assets/icons/pound-sign.svg';
import '../styles/job.scss';
import api from '../api';
import { getToken } from '../auth';
import { UserContext } from '../contexts/userContext';

class Job extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      status: null,
    };

    this.acceptJob = this.acceptJob.bind(this);
    this.completeJob = this.completeJob.bind(this);
    this.declineJob = this.declineJob.bind(this);
  }

  componentDidMount() {
    const { location } = this.props;
    const { status } = location.state;
    this.setState({ status });
  }

  acceptJob() {
    const { location } = this.props;
    const { _id: id } = location.state;
    api.jobs.acceptJob(getToken(), { id }).then((res) => {
      const { status } = res.job;
      this.setState({ status });
    });
  }

  completeJob() {
    const { location } = this.props;
    const { _id: id } = location.state;
    api.jobs.completeJob(getToken(), { id }).then((res) => {
      const { status } = res.job;
      this.setState({ status });
    });
  }

  declineJob() {
    const { location } = this.props;
    const { _id: id } = location.state;
    api.jobs.declineJob(getToken(), { id }).then((res) => {
      const { status } = res.job;
      this.setState({ status });
    });
  }

  render() {
    const { location, history } = this.props;
    const {
      title, date, address, description, pay,
    } = location.state;
    const { status } = this.state;
    const { user } = this.context;
    const formatedDate = new Date(date).toLocaleDateString();
    return (
      <div className="page job">
        <PageHeader title={title} backButton history={history} />
        <h2 className="status">{status}</h2>
        <div className="job-details">
          <ul>
            <li>
              <DateIcon />
              {formatedDate}
            </li>
            <li>
              <LocationIcon />
              {address}
            </li>
            <li>
              <PoundIcon />
              {pay}
            </li>
            <li>
              <NoteIcon />
              {description}
            </li>
          </ul>
        </div>
        {
          user.role === 'model' && (
            <div className="job-actions">
              {status === 'pending' && (
              <>
                <button
                  className="secondary-button"
                  type="button"
                  onClick={this.declineJob}
                >
                                Decline
                </button>
                <button
                  className="primary-button"
                  type="button"
                  onClick={this.acceptJob}
                >
                                Accept
                </button>
              </>
              )}
              {status === 'in progress' && (
              <button
                className="primary-button"
                type="button"
                onClick={this.completeJob}
              >
                            Complete
              </button>
              )}
            </div>
          )
        }
      </div>
    );
  }
}

Job.propTypes = {
  history: PropTypes.shape({}).isRequired,
  location: PropTypes.shape({}).isRequired,
};

Job.contextType = UserContext;

export default Job;
