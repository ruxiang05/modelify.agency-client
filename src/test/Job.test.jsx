/* Uses methods and/or components from react, prop-types, history and enzyme */
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import PropTypes from 'prop-types';
import Job from '../components/Job';
import { UserProvider } from '../contexts/userContext';
import api from '../api';

jest.mock('../api');

const contextTypes = {
  user: PropTypes.object,
};

const modelContext = {
  user: {
    role: 'model',
  },
};
const job = {
  status: 'pending',
  _id: '5c97b9cd0328e6b2aa76a361',
  model: '5c686eb24cb1b33456b0be59',
  agent: '5c686e904cb1b33456b0be58',
  title: 'Test',
  date: '2019-05-26T00:00:00.000Z',
  address: 'address',
  pay: 100,
  description: 'some description',
};

const location = {
  state: job,
};

const agentContext = {
  user: {
    role: 'agent',
  },
};

const history = createBrowserHistory({
  basename: '/',
  keyLength: 0,
});

let wrapper;
beforeAll(() => {
  Job.contextTypes = contextTypes;
});

beforeEach(() => {
  wrapper = mount(
    <UserProvider value={modelContext}>
      <Job location={location} history={history} />
    </UserProvider>,
  );
});

describe('JobCard', () => {
  it('should render JobCard', () => {
    wrapper = shallow(<Job location={location} history={history} />, {
      context: agentContext,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should show action buttons Accept and Decline if user is a model', () => {
    const actionButtons = wrapper.find('.job-actions button');
    expect(actionButtons.length).toBe(2);
    expect(actionButtons.get(0).props.children).toBe('Decline');
    expect(actionButtons.get(1).props.children).toBe('Accept');
  });

  it('should show Complete button after accepting job', async () => {
    const acceptButton = wrapper.find('.job-actions button').at(1);
    await acceptButton.simulate('click');
    expect(wrapper.state().status).toBe('in progress');
    wrapper.update();
    expect(wrapper.find('.job-actions button').text()).toBe('Complete');
  });

  it('should not show any action button after completing the job', async () => {
    const acceptButton = wrapper.find('.job-actions button').at(1);
    await acceptButton.simulate('click');
    wrapper.update();
    const completeButton = wrapper.find('.job-actions button').first();
    await completeButton.simulate('click');
    wrapper.update();
    expect(wrapper.state().status).toBe('complete');
    expect(wrapper.find('.job-actions button').get(0)).toBeUndefined();
  });

  it('should not show any action button after declining the job', async () => {
    const declineButton = wrapper.find('.job-actions button').first();
    await declineButton.simulate('click');
    wrapper.update();
    expect(wrapper.state().status).toBe('declined');
    expect(wrapper.find('.job-actions button').get(0)).toBeUndefined();
  });

  it('should add job to google calendar', () => {
    wrapper.setState({ status: 'in progress' });
    wrapper.update();
    const addToCalendar = wrapper.find('.add-to-calendar button').first();
    addToCalendar.simulate('click');
    expect(api.googleCalendar.addEvent).toHaveBeenCalled();
  });
});
