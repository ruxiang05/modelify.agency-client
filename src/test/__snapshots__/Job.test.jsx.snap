// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should render JobCard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Job
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "state": Object {
          "_id": "5c97b9cd0328e6b2aa76a361",
          "address": "address",
          "agent": "5c686e904cb1b33456b0be58",
          "date": "2019-05-26T00:00:00.000Z",
          "description": "some description",
          "model": "5c686eb24cb1b33456b0be59",
          "pay": 100,
          "status": "pending",
          "title": "Test",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageHeader
          backButton={true}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          title="Test"
        />,
        <h2
          className="status"
        >
          pending
        </h2>,
        <div
          className="job-details"
        >
          <ul>
            <li>
              <ForwardRef />
              5/26/2019
            </li>
            <li>
              <ForwardRef />
              address
            </li>
            <li>
              <ForwardRef />
              100
            </li>
            <li>
              <ForwardRef />
              some description
            </li>
          </ul>
        </div>,
        false,
      ],
      "className": "page job",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": undefined,
          "actionAltText": undefined,
          "actionIcon": undefined,
          "backButton": true,
          "goBackAction": undefined,
          "history": Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          },
          "title": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "pending",
          "className": "status",
        },
        "ref": null,
        "rendered": "pending",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul>
              <li>
                <ForwardRef />
                5/26/2019
              </li>
              <li>
                <ForwardRef />
                address
              </li>
              <li>
                <ForwardRef />
                100
              </li>
              <li>
                <ForwardRef />
                some description
              </li>
            </ul>,
            false,
          ],
          "className": "job-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <ForwardRef />
                  5/26/2019
                </li>,
                <li>
                  <ForwardRef />
                  address
                </li>,
                <li>
                  <ForwardRef />
                  100
                </li>,
                <li>
                  <ForwardRef />
                  some description
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    "5/26/2019",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "5/26/2019",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    "address",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "address",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    100,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  100,
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ForwardRef />,
                    "some description",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "some description",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={true}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            title="Test"
          />,
          <h2
            className="status"
          >
            pending
          </h2>,
          <div
            className="job-details"
          >
            <ul>
              <li>
                <ForwardRef />
                5/26/2019
              </li>
              <li>
                <ForwardRef />
                address
              </li>
              <li>
                <ForwardRef />
                100
              </li>
              <li>
                <ForwardRef />
                some description
              </li>
            </ul>
          </div>,
          false,
        ],
        "className": "page job",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": true,
            "goBackAction": undefined,
            "history": Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            },
            "title": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "pending",
            "className": "status",
          },
          "ref": null,
          "rendered": "pending",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul>
                <li>
                  <ForwardRef />
                  5/26/2019
                </li>
                <li>
                  <ForwardRef />
                  address
                </li>
                <li>
                  <ForwardRef />
                  100
                </li>
                <li>
                  <ForwardRef />
                  some description
                </li>
              </ul>,
              false,
            ],
            "className": "job-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <ForwardRef />
                    5/26/2019
                  </li>,
                  <li>
                    <ForwardRef />
                    address
                  </li>,
                  <li>
                    <ForwardRef />
                    100
                  </li>,
                  <li>
                    <ForwardRef />
                    some description
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ForwardRef />,
                      "5/26/2019",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    "5/26/2019",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ForwardRef />,
                      "address",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    "address",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ForwardRef />,
                      100,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    100,
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ForwardRef />,
                      "some description",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    "some description",
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "user": Object {
        "role": "agent",
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
